FileIO/XmlIO/Boost/BoostXmlGmlInterface.cpp

```diff
@@ -263,8 +262,61 @@ bool BoostXmlGmlInterface::isGmlFile(const boost::property_tree::ptree &root) co
 
 bool BoostXmlGmlInterface::write()
 {
-	INFO ("Writing XML geometry is not implemented here. Please use the Qt XML class for this functionality.");
-	return false;
+	if (this->_exportName.empty()) {
+		ERR("BoostXmlGmlInterface::write(): No geometry specified.");
+		return false;
+	}
+
+	GeoLib::PointVec const*const pnt_vec(_geo_objects.getPointVecObj(_exportName));
+	if (! pnt_vec) {
+		ERR("BoostXmlGmlInterface::write(): No PointVec within the geometry \"%s\".",
+			_exportName.c_str());
+		return false;
+	}
+
+	std::vector<GeoLib::Point*> const*const pnts(pnt_vec->getVector());
+	if (! pnts) {
+		ERR("BoostXmlGmlInterface::write(): No vector of points within the geometry \"%s\".",
+			_exportName.c_str());
+		return false;
+	}
+	if (pnts->empty()) {
+		ERR("BoostXmlGmlInterface::write(): No points within the geometry \"%s\".",
+			_exportName.c_str());
+		return false;
+	}
+
+	// create a property tree for writing it to file
+	using boost::property_tree::ptree;
+	ptree pt;
+
+	// put header in property tree
+	pt.put("<xmlattr>.xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance");
```

Comment:
Since several files in FileIO/XmlIO contain the same output of xml headers,  we may have to get one unique function to do it. This is just a suggestion possibly for another PR.